dockertest/dockercmd.py

```diff
@@ -334,6 +334,49 @@ def execute(self, stdin=None):
                                          stdin=stdin, close_fds=True)
         return self.cmdresult
 
+    def wait_for_ready(self, timeout=None, timestep=0.2):
+        """
+        Monitor the output of a container (including docker logs, in
+        case stdout is detached), waiting for the string 'READY'. Return
+        as soon as we see it. If we don't, throw a meaningful exception.
+
+        :raises DockerExecError: on timeout.
+        """
+        end_time = time.time() + timeout
+        while time.time() <= end_time:
+            time.sleep(timestep)
+            if 'READY' in self.stdout:
+                return
+            # Also check docker logs
+            cid = self.container_id
+            if cid is not None:
+                logs = DockerCmd(self.subtest, 'logs', [self.container_id])
+                logs.execute()
+                if 'READY' in logs.stdout:
+                    return
+
+        # Never saw READY. Did container exit? If so, help user understand why
+        if self.done:
+            msg = "Container exited before READY"
+            if self.exit_status == 0:
+                msg += " (normal exit status)"
+            else:
+                msg += "; exit status = %d" % self.exit_status
+            stderr = self.stderr
+            if stderr:
+                msg += "; stderr='%s'" % stderr
+            raise DockerExecError(msg)
+
+        # Container still running. Must be a timeout.
+        msg = "Timed out waiting for container READY"
+        stdout = self.stdout
+        if stdout:
+            # Juuuuuuust in case it appeared since we last checked
+            if 'READY' in stdout:
+                return
```

Comment:
This doesn't seem right to me.  It presumes the caller cares more about the output than the timeout, which may not be the case.  Otherwise, why stop at double-checking just once?  It's equally likely the string could appear in the microseconds between line 377 and 378.  IMHO, better to ``while time.time() < end_time:`` (since the code is counting "0" seconds elapsed) and leave the double-checking out.  The caller can do it if it's really necessary.